### ISSUE TITLE
Missing signal handlers

### ISSUE THREAD

**Issue by garthk** [equinix-labs/otel-cli#190]
Hitting ⌃C while executing `otel-cli exec sleep 60` prevents sending the span.

<details><summary>Proof-of-concept fix:</summary>

The following demonstrates a fix for the problem on UNIX-style systems:

```diff
diff --git otelcli/exec.go otelcli/exec.go
index 46df894..a560579 100644
--- otelcli/exec.go
+++ otelcli/exec.go
@@ -5,8 +5,10 @@ import (
        "fmt"
        "os"
        "os/exec"
+       "os/signal"
        "runtime"
        "strings"
+       "syscall"
 
        "github.com/spf13/cobra"
        tracepb "go.opentelemetry.io/proto/otlp/trace/v1"
@@ -59,7 +61,10 @@ func doExec(cmd *cobra.Command, args []string) {
                shell = "/bin/sh"
        }
 
-       child := exec.Command(shell, "-c", commandString)
+       ctx, done := signal.NotifyContext(cmd.Context(), syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM)
+       defer done()
+
+       child := exec.CommandContext(ctx, shell, "-c", commandString)
 
        // attach all stdio to the parent's handles
        child.Stdin = os.Stdin
```

I think a full solution would also call `done` after `child.Run` to close that context, and call `signal.NotifyContext` again to get a context to pass to `SendSpan`.
</details>

**Comment by tobert** [comment]
Hmm, your PoC looks right. The remaining thing to figure out is how to test it. Not terribly hard, just gonna need some additions to the test framework.

**Comment by tobert** [comment]
Fixed in #272.  I tried using `exec.CommandContext` and couldn't get the behavior right. It's worth revisiting eventually but for now ctrl-c should behave.
